zhusuan/distributions/multivariate.py

```diff
@@ -29,6 +31,144 @@
 ]
 
 
+class MultivariateNormalTriL(Distribution):
+    """
+    The class of multivariate normal distribution, where covariance is
+    parameterized with the lower triangular matrix in Cholesky decomposition,
+
+        .. math :: L \\text{s.t.} LL^T = \Sigma.
+
+    See :class:`~zhusuan.distributions.base.Distribution` for details.
+
+    :param mean: An N-D `float` Tensor. mean[..., i_{N-1}, :] corresponds to a
+        single distribution.
+    :param cov_tril: An (N+1)-D `float` Tensor. Each slice represents the lower
```

Comment:
If put into the context of Gaussian process, there could be cases that a covariance function instead of a covariance matrix is needed. Although you could first figure out the dense covariance matrix and then passed to the MultivariateNormal dsitribution, this is often not the case in general computation. How could we support that?

Reply:
Not sure as it's been a long time, but I guess in sparse VGP they just calculate the covariance (in batches) and does cholesky. For Gaussian likelihood there are [numerical issues](http://gpflow.readthedocs.io/en/latest/notebooks/SGPR_notes.html) but in the general case it might work. I'll check if this works soon. 

I don't know if there are other solutions. And it's possible that in the end it only helps small-scale GPs like [the example in Edward](https://github.com/blei-lab/edward/blob/master/examples/cox_process.py).